Add WSL support file template

Vanco opened the pull request.

ikeycode committed:
Initial commit

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm: Add basic scripts to handle LiveOS portion of img construction

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/img: Remove moss downloads, nuke dirty tree

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
Stash more changes for testing..

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
Add gitignore

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
Instructions

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/pkglist: Cleanup

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/img: Hardlink internal initrd

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/pkglist: Add `ps`

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
live-os.conf: No more serial hax, we have tty now

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/img.sh: hybrid ISO correctly configured, XZ for size

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
img.sh: Initial productisation cleanup

- Add license headers for compliance
- Check that script is run from correct directory up front
- Check for necessary tools up front
- Be verbose and forceful about cleanups and moves/copies

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
img.sh: systemctl is not a host prerequisite

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
kvm/pkglist: Include intel/i915 for light hardware testing

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
Update copyright ranges

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm: Use overlayfs to form the throwaway dracut image fully

This allows us to access binutils etc in chroot without contaminating the
root image.

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
update gitignore for overlayfs

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/img: Switch squashfs to zstd, xz just toooo slooooww

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/img: Shrink image

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/img: Remove not relevant comments

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/img: Update hwdb to speed up boot

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/pkglist: neofetch for good measure

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/img: Enlargen for LTO..

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm: Require `overlayfs` redirect_dir option for new moss

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm: Add testing script for generating qemu installs

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
diskImg: Use systemd/ path on ESP to match bootctl

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
diskImg: Use proper paths

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
diskImg: Create snekdir

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
diskImg: More conformity stuff

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
installed-os: Use namespace

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
diskImg: Fix bug in systemd units complaining on startup

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm: There is only volatile now \o/

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
pkglist: Drop intel firmware (no i915) and add missing grep

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/pkglist: Require setfont

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Lightly fork kvm, we'll do most of our work here now

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/initrdlist: Sort. Irritating

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/initrdlist: Install firmware during initrd stage to avoid pollution

We get microcode behaviour for "free" by using the overlayfs to avoid polluting
the target image

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/img.sh: Fix crappy copy paste

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/img: Ensure we build the microcode

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/img: Increase efi.img size to accomodate farkin massive initrd

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
Port scripts to moss-rs

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
installed-os: Better title

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm/diskImg: Prepare as we would the live ISO, fixing UEFI console issues

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
Add split design for package lists for convenience

- Add `pkglist-base` containing common tools for moss-managed roots
- Add `create-sosroot.sh` for creating systemd-nspawnable `sosroot/` dir
- Tweak desktop/ and kvm/ `img.sh` scripts for the split pkglist design
- Add README, firmware and launch script for testing Qemu `kvm/disk.img`
- Update README.md with invocation examples
- Switch to MPL-2.0 License

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
Merge pull request #1 from serpent-os/split-pkglists

Add split design for package lists for convenience

ermo committed:
Add binary(tar) to pkglist-base for convenience

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
Generate script for booting into sosroot/ later on

Particularly useful if specifying a custom directory for a sosroot/ dir.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
.gitignore: Ignore generated container boot script

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
create-sosroot.sh: Fix typo in createBootScript()

Signed-off-by: Rune Morling <ermo@serpentos.com>

der-eismann committed:
Add script to create virt-manager VM

ermo committed:
image scripts: [NFC] Rename pkgs var to PACKAGES

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
create-virtio-vm: List correct override in help text

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
virt-manager: Tweak template to work on older libvirt

- Use q35 model instead of pc-q35-8.1
- Remove iTCO watchdog from config
- Use 4 cores by default (2 is too little for serpent)

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
nspawn and virt-manager consolidation and tweaks

- create-virtio-vm.sh script now uses UEFI instaed of BIOS
- basic-setup.sh script with shared setup and cleanup functions added
- basic-setup.sh script sets up an empty root passwd by default
- basic-setup.sh sets up a nss-systemd compatible /etc/nsswitch.conf ootb
- create-virtio-vm.sh and create-sosroot.sh scripts uses basic-setup.sh

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
basic-setup: Drop /etc/nsswitch.conf creation

It turns out that moss triggers did not correctly respect the dependency
order specified in the triggers, making it necessary to create
/etc/nsswitch.conf manually before attempting to install systemd.

Systemd is supposed to create /etc/nsswitch.conf itself first thing if
it doesn't already exist, but due to the aforementioned bug it didn't,
which in turn made moss error out with missing user/group errors.

However, this is now fixed, so the workaround here is no longer needed.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
virt-manager-vm: Add liberation-fonts-ttf ootb

Helps avoid Cairo segfaults when installing GDM and GNOME for now.

Will get fixed properly in the future.

Signed-off-by: Rune Morling <ermo@serpentos.com>

flippedbits committed:
Allow custom vm name. Argument to display help message.

ermo committed:
virt-manager-vm: Tweak creation script.

- Create mount point for host /var/cache/boulder in basic-setup script
- Unmount and re-mount host /var/cache/boulder into guest as bind mount
- Ensure that script work on fedora, were edk2-ovmf installs multiple
  versions of firmware in different folders
- Tweak message text slightly

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
basic-setup: Create ${BOULDERCACHE} dir

Signed-off-by: Rune Morling <ermo@serpentos.com>

flippedbits committed:
Add case insensitive flag to grep to allow finding ovmf fd in debian based systems.

ikeycode committed:
Merge pull request #5 from flippedbits/fix-case-sensitive-grep-for-ovmf

Add case insensitive flag to grep

ikeycode committed:
desktop: Add some GNOME to the pkglist

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Tackle obesity crisis by providing a larger belt

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
desktop: Use tmpfs for ephemeral storage and compress w/lz4 (#6)

* desktop: Add workdir prefix to paths, clean up mounts

Signed-off-by: Rune Morling <ermo@serpentos.com>

* desktop: Speed up ISO generation a bit (tmpfs)

This approach avoids the OverlayFS terminal slowness, so should make
image blitting a fair bit faster than before when creating ISOs.

Previous times were around 8m30s.

Current time is about 1m30s.

The tmpfs approach requires 8GiB free RAM to work.

Signed-off-by: Rune Morling <ermo@serpentos.com>

* desktop: Clean up variable naming for readability

Signed-off-by: Rune Morling <ermo@serpentos.com>

* desktop: Add README.md w/troubleshooting tips

Signed-off-by: Rune Morling <ermo@serpentos.com>

* desktop: Add comments to mksquashfs options

Signed-off-by: Rune Morling <ermo@serpentos.com>

* desktop: Small tweaks, address review nits

- Fail earlier if not run as root
- Switch from kebab-case to snake_case
- Add comments w/recommendations re. squashfs settings (prod vs. testing)
- Remove no longer relevant and commented out code

Tried testing the tar invocation against rsync -Hal source/ target -- to
my surprise, tar achieved a speed of ~175MB/s whereas rsync achieved only
~150MB/s. Therefore, the tar invocation was kept.

Signed-off-by: Rune Morling <ermo@serpentos.com>

* desktop: mksquashfs supports a folder as source

This change cuts the ISO build time from ~1m30s to ~1m10s on my system.

Signed-off-by: Rune Morling <ermo@serpentos.com>

* desktop: Fix squashfs.img generation + cleanups

The previous version mistakenly added the serpentfs/ prefix to the root.
This commit fixes that and ensures the output shows the contents of the
new root, so mksquashfs sources can be set up accordingly.

Signed-off-by: Rune Morling <ermo@serpentos.com>

---------

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
desktop: Update to use new moss --cache feature

Has the potential to shave a bit of time off successive image generation
runs. On my system, the times went from 1m18s (cold first run, uncached)
to 1m5s (warm, cached).

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
desktop: Use newly landed moss rollback feature

Prior to this (without rollback), build took 1m1s to 1m5s on my system.

Using the rollback feature, build takes ~45s on my system.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
desktop: Use gnome-console, drop liberation fonts requirement

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
virt-manager-vm: Drop liberation-fonts-ttf, add networkmanager

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
Update README, add xz compression for desktop/ ISO

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
README.md: Add missing mystery word

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
desktop/pkglist: Pimp the offerings

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/pkglist: Add font-hack for monospace

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
kvm: Nobody likes this

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
scripts: Tweak defaults and use new moss flags

- Update to use moss --cache funcitonality
- Add -y to moss invocations
- Switch the default dir to be the fully qualified path to sosroot in the
  current folder for both virtio-vm and nspawn

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
virt-manager-vm: Use OVMF_CODE.fd as OVMF.fd hangs

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
desktop/pkglist: Add some missing packages

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
pkglist-base: Remove explicit moss-container (deprecation time)

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
desktop: Replace moss-container with systemd-nspawn

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
README.md: Use desktop/ instead of kvm/

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
pkglist-base: Add tzdata package

This shuts up systemd-firstboot runs complaining about /etc/localtime.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
pkglist-base: Remove tzdata as it is now a rundep

... of systemd proper, as it should be.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
virt-manager: Set default console font to ter-v32n

This needs enabling in both kernel config and in dracut config to work
properly.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
virt-manager-vm: Tweak kernel command line

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
desktop: Quote all variables to enable using spaces

Thank you to @QORTEC for the report.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
virt-manager: Set default name to serpent_virtiofs

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
basic+vm: Tweak script quoting and prereqs handling

The new onboarding `just get-started` target installs moss to a
`/usr/local` prefix by default.

All script path variables should now be quoted as well.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
basic+vm: Test moss and print its version on run

This should make it easier to confirm that the system is using the
expected version of moss from e.g. onboarding or the system install.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
basic-setup: Update to newest moss onboarding guide

This will make basic-setup.sh work again.

ermo committed:
Revert "basic-setup: Update to newest moss onboarding guide"

This reverts commit 0878109f980b836fb462c0cd64f323465aa5c77d.

ermo committed:
basic-setup: Make it work again

- Ensure that /var/cache/boulder exists
- Enable the local repo from within the new root

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
virt-manager-vm: Update kernel, display boot params

The linux-kvm kernel now uses the blsforme spec (paths and boot.json),
which required an update to the virt-manager template.

In addition, the default display size is now 1440x900 after KMS is
initialised at boot, which will hopefully make the systemd-firstboot
experience slightly less bad.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
desktop: Add lichen to pkglist

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/img.sh: Fix installation of kernel

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
desktop: Selectabled compression, remove quiet arg

Compression can be specified with `sudo -E COMPRESSION=foo ./img.sh`

Valid compression types are:
- lz4 (default, quick to build, good for spotting size regressions)
- lz4hc (use zstd3 instead)
- zstd3 (recommended if you're going to write an image to a USB stick)
- zstd19 (recommended for release images)
- xz (provided as reference for zstd19)

I removed the `quiet` argument from live-os.conf because I want to see
what the ISO is attempting to do when initialising my hardware for
troubleshooting purposes. I would recommend that we don't re-add `quiet`
until we're out of pre-alpha...

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
pkglist-base: `bash` now provides `/bin/sh` these days

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/initrd-list: Add dash here..

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
pkglist-base: Loose weight: Sue me

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/README: Add help for Serpent boot

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/pkglist: Add serpent-gnome-defaultso

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
pkglist-base: Explicitly require ca-certitificates not nss

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/img: Add live user, fix root perms everywhere

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
README: Give usable qemu CLI

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Kill live user password

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/img: Derp

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Add firefox in

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/img: Fix groups

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/live-os.conf: Update to prealpha0 branding

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/pkglist: Fix fonts

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
desktop: Use MiB for informational ISO size output

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
desktop/rootfs_extra: Force a bashrc in to introduce Lichen

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
desktop: Add various console troubleshooting utils

The idea is to make it easier to gain insight into hardware support and
network configuration from within the live ISO.

Depends on https://github.com/serpent-os/recipes/pull/247

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
desktop/img: Moar larger kernel

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

Nivalux committed:
desktop: fixed sudo for live user

ermo committed:
desktop: Use getopts to parse compression type

In addition, add more useful help text output, including suppported
compression types, default compression type and a recommendation to use
zstd3 for quick turnaround times for local testing.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
desktop: Update README.md

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
desktop: Add cosmic_pkglist

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/pkglist: Add spice-vdagent

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
Use personal boulder cache, set up nspawn script

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
virtio-vm: Pick first UEFI firmware payload found

On systems with more than one suitable and valid path to OVMF_CODE.fd,
the script would error out.

This commit fixes that by always selecting the first valid OVFM_CODE.fd
payload found.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
desktop: Require gnome-disk-utility in the ISO

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
img: Fat ass

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
desktop: Fix end-of-run image type help text

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
desktop/pkglist: he who shall not be named

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Use quiet boot now

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Integrate plymouth splash

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Add starship by default

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Totally innoculous change

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
Add justfile so we don't need to remember stuff

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
Rework so building is as simple as `just flavor=cosmic`

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
img: Smart cookie time: Precalculate required efi.img size

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
Use the new volatile repo URI for image production

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
justfile: use sdl, gtk display is so broken

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
justfile: Disable SDL cursor to avoid severe confusion

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/cosmic: Add gnome-backgrounds..

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/img: Build initrd with plymouth

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/gnome: Add starship to ISO

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
justfile: Add help, autogenerate output iso name

This should significantly improve the ootb experience for people creating
isos.

Also add a convenient release recipe, which will produce both zstd19
compressed `gnome-snek-zstd19.iso` and `cosmic-snek-zstd19.iso` images.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
desktop: Add -o output option, tweak error handling

The ISO will be named `<output>.iso`. This is useful when generating ISO
output filenames based on flavor and compression in the justfile.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
desktop/gnome: Add gparted to ISO

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
desktop/cosmic: Add plymouth to cosmic_pkglist

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
Merge pull request #11 from serpent-os/feat/add-justfile-cmdline-options

Upgrade the justfile experience for generating ISOs a bit

ermo committed:
justfile: Drop sudo for ls in _list-isos recipe

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
desktop: Add sed hack for uutils coreutils 'ls'

The ls from uutils coreutils does not add the size suffix next to the
block size when using e.g. 'ls -s --block-size=M' as the invocation.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
pkglist-base: Finally swap neofetch for fastfetch

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Organise gnome defaults, flesh it out a bit

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
Remove spaces from pkglists, replace gnome-system-monitor with resources

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
justfile: Add memory stanza to 'boot' help example

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
desktop/gnome: Require file-roller for le file rolling

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Flesh out the offering

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Add livecd polkit rules

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/live-os.conf: Not so secret santa

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/gnome_pkglist: Add joeys latest

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/img: Remove hacks that are taken care of by triggers now

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
dekstop/img: Stop forcing getty1, presets take care of it

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Fix cmdline for radeon and exclude DM/MD/RAID/LVM/LUKS

This should fix an number of bootability issues we're having.

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop: Use stashed isolinux assets for compliant ISO structure

It'll deliberately fail to boot on non UEFI, and we can get folks to
show us the screenshot for the obvious failure. Apart from that it fixes
the structure now to work with etcher and potentially other firmwares.

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
desktop/gnome: Add gnome-calendar

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ikeycode committed:
justfile: Add `-cpu host` so x86_64-v2 works..

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

ermo committed:
README: Tweak to rely on `justfile` for help

Signed-off-by: Rune Morling <ermo@serpentos.com>

ikeycode committed:
desktop/img: Forcibly refresh flatpak

Still a bit buggy, flatpak being flatpak.

Signed-off-by: Ikey Doherty <ikey@serpentos.com>

Biswa96 committed:
Fix typo with BOULDERCACHE variable

ikeycode committed:
desktop: Modify branding for AerynOS

Signed-off-by: Ikey Doherty <ikey@aerynos.com>

ikeycode committed:
desktop: Fix strings in ISO itself

Signed-off-by: Ikey Doherty <ikey@aerynos.com>

ikeycode committed:
desktop/img: Rework for AerynOS

Signed-off-by: Ikey Doherty <ikey@aerynos.com>

ikeycode committed:
feck off sneks

Signed-off-by: Ikey Doherty <ikey@aerynos.com>

ikeycode committed:
desktop/img: Use the raw volatile (dotdev)

Signed-off-by: Ikey Doherty <ikey@aerynos.com>

ikeycode committed:
desktop: Fix pkglist selections

Signed-off-by: Ikey Doherty <ikey@aerynos.com>

ikeycode committed:
desktop/gnome: No need for zed in live ISO really - its massive.

Signed-off-by: Ikey Doherty <ikey@aerynos.com>

ikeycode committed:
desktop: Update versioning

Signed-off-by: Ikey Doherty <ikey@aerynos.com>

ikeycode committed:
desktop: Cleanup jank

Signed-off-by: Ikey Doherty <ikey@aerynos.com>

ikeycode committed:
Account for future jank.

Signed-off-by: Ikey Doherty <ikey@aerynos.com>

ermo committed:
desktop: Refactor, add prompt to continue w/-y arg

- Add -t TMPDIR option as some distros need /var/tmp (defaults to /tmp)
- Add continue prompt that shows status for ISO and asks to continue
- Add override '-y' option for ^ that continues without asking
- Add final_cleanup function that always runs
- Add trapexit clause that calls the final_cleanup function
- Cleanup now checks for valid TMPFS variable to avoid hosing / (!)
- Make the default package list actually work (selects gnome_pkglist)
- Make the '-?' option actually work
- Make the actual "build the ISO" steps run as a callable function

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
scripts: Switch from Serpent OS to AerynOS branding

Temporarily switch to infratest.aerynos.dev as the download server.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
desktop: Refactor shared components to desktop-base

Enable the pkg lists to contain empty lines and comment-only lines. These
will get removed prior to parsing the files to bash arrays.

With this commit, both the GNOME and the COSMIC ISOs build and boot.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
desktop: Add binary(zstd) to initrdlist

Prior to this addition, dracut gave this error:

```
dracut[E]: Cannot execute compression command 'zstd', falling back to default
dracut[I]: Using auto-determined compression method 'gzip'
```

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
Switch to build.aerynos.dev/volatile repository

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
README: Document invocation via `just` recipes

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
desktop: Comment out hardcoded local_repo

Can be later uncommented for users that need. They would obviously need
to user their own local_repo path.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
live-os.conf: Update version to 2025.07

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
basic-setup: Fix systemd-nspawn container setup

- Add normal 'live' user with full sudo access and no passwd
- Ensure 'root' is set up to use no password
- Update pkglist-base to rely on pkgset-aeryn-{base,utilities} + sundry

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
desktop: Update pkg lists based on pkgset-aeryn-*

This should make ISO desktop environment configurations easier to
configure from the recipe repo via repo pkgset-aeryn-* changes going
forward.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ReillyBrogan committed:
fix: Use pkgsets and ensure that all drm drivers are installed

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>

ermo committed:
justfile: Update path to edk2 firmware

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
README: Improve example invocations

ermo committed:
desktop: Update image version to 2025.08

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
virt_manager_vm: Update to AerynOS branding

Switch to pkgset-aeryn-sway-minimal pkgset for the ENABLE_SWAY option.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
virt-manager-vm: Update template to linux2024

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
virt-manager-vm: Tweak kvm-related pkglist

Switch to linux-desktop until the KVM-only kernel is brought up to speed.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ReillyBrogan committed:
Add sof-firmware

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>

Vanco committed:
Add WSL support file template

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AerynOS/img-tests?pullRequest=24) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/AerynOS/img-tests?pullRequest=24) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AerynOS/img-tests?pullRequest=24) it.</sub>

ermo commented:
Out of interest, was any of this generated using AI tools? This would change how we review your contribution.

Second, have you had a look at the script that generates a root that can run via systemd-nspawn? Perhaps the present guide is not really necessary, if the wsl form of the script mirrors the way we generate the systemd-nspawn root?

Vanco commented:
@ermo 
The initial idea of creating the WSL version originated from the fact that I only had an Intel laptop and an M2 Mac mini. I really liked AerynOS and wanted to install it in WSL. Starting from the AerynOS-2025.08-GNOME-live-x86_64.iso I obtained, I created a WSL Root FS step by step with the help of AI. Essentially, it is the squashfs of a Live CD. After successfully importing it, I let AI summarize the complete steps in Chinese and had it translated into English. This is how the Guide came to be. You could say it was AI-generated.

I think if the official version of WSL could be released, it would greatly help in promoting AerynOS.

After I posted this guide in the discussion area, I adopted your suggestions and am preparing to submit a PR.

I think if it's published on img-tests, it may no longer be necessary to convert from Live CD to WSL's rootfs, and it should be a more direct way to build a WSL distribution from a Shell script.

However, I know very little about AerynOS internally, so I may not be able to complete this task. As you mentioned, I will share my ideas for the Guide that I can provide, as well as the configuration files needed to create a WSL distribution, which I have extracted from the Microsoft official website. These files have either been practiced in my manual steps or are essential for producing the WSL distribution.These documents were written by me by hand, not generated by AI.

I think this is probably all I can do. I hope it helps you a little bit in creating the WSL distribution version. If you find it useless, you can also ignore this PR.

ermo commented:
Thank you for the summary.

We already know how to create the squashfs, so this will likely be a "nicer" process if we do build it directly, rather than trying to do it from the ISO.

Just be aware that AerynOS is currently a "Tech Preview", is in Alpha and has some significant limitations on the package management and infrastructure side that we are working towards eliminating.

ermo committed:
desktop: Update version to 2025.10

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
desktop: Write name-version-release (.nvr) file

This will enable e.g. distrowatch and other aggregation site to list our
ISO package versions.

Signed-off-by: Rune Morling <ermo@serpentos.com>

ermo committed:
desktop: Update live-os.conf to 2025.12

The 2025-12-09 ISO was generated with this, the change just wasn't
committed before now.

Signed-off-by: Rune Morling <ermo@aerynos.com>

Vanco committed:
Merge branch 'AerynOS:main' into main

NomadicCore committed:
desktop: Update live-os.conf to 2026.01

Update the version in preparation for a 2026.01 ISO release

NomadicCore committed:
Repo: Switch to new stream url's for volatile repo

ermo committed:
desktop: Add show-versions.bash script

This script will print the versions of a list of what was considered
relevant packages at the time of its last update.

Provided as-is and probably has "undocumented features".

Signed-off-by: Rune Morling <ermo@aerynos.com>

ermo committed:
desktop/show-versions.bash: Add wayland components

Signed-off-by: Rune Morling <ermo@aerynos.com>

ermo committed:
desktop: Update live-os.conf to 2026.02

Signed-off-by: Rune Morling <ermo@aerynos.com>

CookieSource committed:
Add BIOS error menu assets

CookieSource committed:
Update desktop/isolinux.cfg

Co-authored-by: Rune Morling <rune.morling+github@proton.me>

NomadicCore committed:
desktop: Update live-os.conf to 2026.03

CookieSource committed:
Add credits for the bios image

ermo committed:
BIOS boot: Tweak screen text for balance

Add `just biosboot` option to enable quick testing of what the BIOS boot
screen will look look when an ISO has been built.

Signed-off-by: Rune Morling <ermo@aerynos.com>

NomadicCore committed:
show-versions.bash: Change mangowc to mangowm

Vanco committed:
Merge branch 'AerynOS:main' into main

Vanco committed:
Add script to build WSL distribution.

Vanco committed:
Update build-wsl.sh

Vanco committed:
add img.sh for wsl

Vanco committed:
remove alias.

Vanco committed:
img.sh

Vanco committed:
img.sh

Vanco committed:
Add build-wsl

Vanco committed:
remove guide.md.

Vanco committed:
remove guide.md.
